qt/widgets/instrumentview/src/InstrumentActor.cpp

```diff
@@ -165,45 +205,47 @@ void InstrumentActor::setUpWorkspace(
   // set the ragged flag using a workspace validator
   auto wsValidator = Mantid::API::CommonBinsValidator();
   m_ragged = !wsValidator.isValid(sharedWorkspace).empty();
-
-  /// Keep the pointer to the detid2index map
-  m_detid2index_map = sharedWorkspace->getDetectorIDToWorkspaceIndexMap();
 }
 
-/** Used to set visibility of an actor corresponding to a particular component
-* When selecting a component in the InstrumentTreeWidget
-*
-* @param visitor :: Visitor to be accepted bu this actor.
-* @param rule :: A rule defining visitor acceptance by assembly actors.
-*/
-bool InstrumentActor::accept(GLActorVisitor &visitor, VisitorAcceptRule rule) {
-  bool ok = m_scene.accept(visitor, rule);
-  visitor.visit(this);
-  invalidateDisplayLists();
-  return ok;
+void InstrumentActor::setupPhysicalInstrument() {
+  auto sharedWorkspace = getWorkspace();
+  Mantid::Kernel::ReadLock _lock(*sharedWorkspace);
+
+  if (isPhysicalView()) {
+    auto instr = sharedWorkspace->getInstrument()->getPhysicalInstrument();
+    if (instr) {
+      m_isPhysicalInstrument = true;
```

Comment:
Very minor point, but I would move the setting of this flag to the last statement in the block to give stronger exception guarantees.